### PR TITLE
Added Parser and Tests / Update from Powercloud GmbH

### PR DESCRIPTION
**Pull request by twitnic**
Hey,

that's the current codebase from our company Powercloud GmbH from Germany.
We have internally "updated" this package step by step. and we would like to give this updates back to the official repo.

So, what do you think. Its a big PR, but its many new cool code, too.

**Added Parser for:**

* Commerzbank
* Deutsche Bank
* German Bank
* Landesbank Berlin
* NuaPay
* Oldenburgische Landesbank
* Sparkasse
* StarMoney
* UniCredit

**Added additional Methods in Transaction:**

* getCode
* getRef
* getBankRef
* getGVC
* getTxText
* getPrimanota
* getExtCode
* getEref
* getBIC
* getIBAN
* getAccountHolder
* getKref
* getMref
* getCred
* getSvwz

### CODE REVIEW

**Comment by twitnic** [comment]
@sandermarechal what did you say?

**Comment by sandermarechal** [comment]
Hi @twitnic, this looks pretty good! A few small comments but that is all (see the inline code reviews).

You seem to be using this library quite a lot. Personally I don't use it anymore since the project I created this for never came off the ground. Since you are much more active than I am with this library, would you like to take it over as maintainer perhaps?

**Comment by twitnic** [comment]
Hey @sandermarechal ,

I didnt see any comments? :/

Yes. I address it internally. Maybe our firm could be the maintainer or myself.
We work every day with this package. And would like to share our developments with each other. :)

**Comment by sandermarechal** [comment]
Whoops, looks like the comments only show up after submitting the entire review. They should show up now :-)

File: lib/Jejik/MT940/Exception/UserException.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Licensed under the MIT license
+ *
+ * For the full copyright and license information, please see the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Jejik\MT940\Exception;
+
+/**
+ * Class NoParserFoundException An exception that is thrown when no one of the given parsers accepted the document.
```
[the remark this answers is not in the document]
Comment (twitnic) [reply lib/Jejik/MT940/Exception/UserException.php]: Fixed. Thanks.

File: lib/Jejik/MT940/Reader.php [hunk 2]
```diff
@@ -23,64 +23,95 @@ class Reader
...
      */
     private $defaultParsers = array(
-        'ABN-AMRO'    => 'Jejik\MT940\Parser\AbnAmro',
-        'ING'         => 'Jejik\MT940\Parser\Ing',
-        'Knab'        => 'Jejik\MT940\Parser\Knab',
-        'PostFinance' => 'Jejik\MT940\Parser\PostFinance',
-        'Rabobank'    => 'Jejik\MT940\Parser\Rabobank',
-        'Sns'         => 'Jejik\MT940\Parser\Sns',
-        'Triodos'     => 'Jejik\MT940\Parser\Triodos'
+        'ABN-AMRO'    => \Jejik\MT940\Parser\AbnAmro::class,
```
[the remark this answers is not in the document]
Comment (twitnic) [reply lib/Jejik/MT940/Reader.php]: fixed. Thank you.

File: lib/Jejik/MT940/Parser/AbnAmro.php [hunk 3]
```diff
@@ -98,4 +98,12 @@ protected function contraAccountName(array $lines): ?string
 
         return null;
     }
+
+    /**
+     * Get an array of allowed BLZ for this bank
+     */
+    public function getAllowedBLZ(): array
```
[the remark this answers is not in the document]
Comment (twitnic) [reply lib/Jejik/MT940/Parser/AbnAmro.php:105]: What did you say to `bankCode` oder `bankCodeNumber`?

**Comment by sandermarechal** [comment]
Perfect, thank you!

**Pull request merged**